### DISCUSSION TITLE
MSBuild executable

### DISCUSSION THREAD

**Discussion by asbjornu** [dotnet/source-build#2769] [Q&A]
With https://github.com/Homebrew/homebrew-core/pull/92991 implemented, I'm noticing when running [JetBrains Rider](https://www.jetbrains.com/rider/) and the [`jb` CLI](https://www.jetbrains.com/help/resharper/CleanupCode.html) that the location of MSBuild is not detected. I've tried to create `/usr/local/bin/msbuild` with the following contents without luck:

```bash
#!/bin/bash
exec "/usr/local/bin/dotnet" "/usr/local/Cellar/dotnet/6.0.100/libexec/sdk/6.0.100/MSBuild.dll" "$@"
```

Does an environment variable such as [`MSBuildSDKsPath` mentioned here](https://siderealdailymentioner.com/manjaro-dotnet-core-rider/) need to be created, or something else saying to Rider and other tools where MSBuild is?

**Comment by omajid** [comment]
I am not familiar with Rider, but does it log/display any errors that might provide hints?

Would it be possible to use `dtruss`/`strace` or maybe even an `echo "$@" >> $HOME/save-rider-caller-args.log` in your script to figure out how Rider tries to invoke msbuild?

Maybe try `exec dotnet msbuild "$@"`?

**Reply by asbjornu** [reply]
I've managed to manually configure MSBuild's placement in Rider, but since the `jb` CLI is also not auto-discovering MSBuild's placement, I thought perhaps there was a standard environment variable these tools were looking for which the .NET installer is setting and the Homebrew formula isn't.
